File: THCS/2023_2024/Hue_2324_Phong/Bai1/BAI1.py
def chia_het_hoan_toan(num):
    num_str = str(num)
    for digit in num_str:
        if digit == '0' or num % int(digit) != 0:
            return False
    return True

def dem_so_chia_het_hoan_toan(arr):
    ket_qua = [num for num in arr if chia_het_hoan_toan(num)]
    return len(ket_qua), ket_qua

# Đọc dữ liệu từ file
with open('SOCHIAHET.INP', 'r') as f:
    n = int(f.readline().strip())
    arr = list(map(int, f.readline().strip().split()))

# Tính toán
count, result_list = dem_so_chia_het_hoan_toan(arr)

# Ghi kết quả ra file
with open('SOCHIAHET.OUT', 'w') as f:
    f.write(f"{count}\n")
    f.write(" ".join(map(str, result_list)))

File: THCS/2023_2024/Hue_2324_Phong/Bai1/BAI1.cpp
#include <iostream>
#include <vector>
#include <fstream>
using namespace std;

// Hàm kiểm tra số chia hết hoàn toàn
bool chiaHetHoanToan(int num) {
    int original_num = num;
    while (num > 0) {
        int digit = num % 10;
        if (digit == 0 || original_num % digit != 0) {
            return false;
        }
        num /= 10;
    }
    return true;
}

int main() {
    ifstream infile("SOCHIAHET.INP");
    ofstream outfile("SOCHIAHET.OUT");

    int n;
    infile >> n;
    vector<int> arr(n);

    for (int i = 0; i < n; ++i) {
        infile >> arr[i];
    }

    vector<int> ketQua;
    for (int num : arr) {
        if (chiaHetHoanToan(num)) {
            ketQua.push_back(num);
        }
    }

    outfile << ketQua.size() << endl;
    for (int num : ketQua) {
        outfile << num << " ";
    }
    outfile << endl;

    infile.close();
    outfile.close();
    return 0;
}
